File: BOJ/코딩테스트 시작을 위한 백준 문제 추천/10828.py
from sys import stdin
input = stdin.readline

n = int(input())
s = []

for _ in range(n):
    cmd = input().split()
    
    if len(cmd) == 2:
        cmd[1] = int(cmd[1])
        s.append(cmd[1])
    elif cmd[0] == 'pop':
        print(s.pop()) if s else print(-1)
    elif cmd[0] == 'size':
        print(len(s))
    elif cmd[0] == 'empty':
        print(int(not s))
    else:
        print(s[-1]) if s else print(-1)

File: 바킹독의 실전 알고리즘/0x05강 스택/10828.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((line) => line.split(" "));

const n = Number(input[0][0]);
const stack = [];
let answer = "";

for (let i = 1; i <= n; i++) {
  const command = input[i];
  if (command.length > 1) {
    const item = Number(command[1]);
    stack.push(item);
  } else {
    const size = stack.length;
    switch (command[0]) {
      case "pop":
        answer += size === 0 ? -1 : stack.pop();
        break;
      case "size":
        answer += size;
        break;
      case "empty":
        answer += size === 0 ? 1 : 0;
        break;
      case "top":
        answer += size === 0 ? -1 : stack[size - 1];
        break;
    }
    answer += "\n";
  }
}

console.log(answer);
